src/splinterdb.c

```diff
@@ -631,6 +631,43 @@ validate_key_length(const splinterdb *kvs, uint64 key_length)
    return 0;
 }
 
+/*
+ * Validate that a key being inserted is within [min, max]-key range.
+ */
+bool
+splinterdb_validate_key_in_range(const splinterdb *kvs, slice key)
+{
+   const data_config *cfg = kvs->shim_data_cfg.app_data_cfg;
+
+   int cmp_rv = 0;
+
+   // key to-be-inserted should be >= min-key
+   cmp_rv = cfg->key_compare(
+      cfg, slice_create(cfg->min_key_length, cfg->min_key), key);
+   if (cmp_rv > 0) {
+      platform_error_log("Key '%.*s' is less than configured min-key '%.*s'.\n",
+                         (int)slice_length(key),
+                         (char *)slice_data(key),
```

Comment:
Keys are not C strings.  

Use the data_config API for printing keys.

Reply:
Even better, use the data_internal.h API for printing the keys, i.e.
```
platform_error_log("... %s ... ",
                               ...,
                               key_string(data_cfg, key),
                               ...);
```
where `key` is a slice.

Reply:
@rtjohnso - Reworked as suggested.

(I'm not entirely sure that printing keys in hex-dump will aid debugging. Most common keys would likely be strings, so '%.*s' was quite helpful. Anyway -- we can re-visit this later on.)

Kindly re-review the delta-diff commit uploaded.

Reply:
I don't think this is a matter of taste.  It is a matter of not crashing the program.

Keys are not C strings.  They are not required to be null terminated or even contain any null bytes at all.  Passing them to a printf-style function as a "%s" is a straight-up SEGFAULT waiting to happen.

Furthermore, the appliction gets to define how keys are printed.  If they don't want hex dumps, they can define a custom function for how to print keys.

Reply:
OK, point noted & agreed. Thanks for catching this sloppy code.